scene/resources/animation.h

```diff
@@ -45,7 +45,7 @@ class Animation : public Resource {
 
 	static inline String PARAMETERS_BASE_PATH = "parameters/";
 
-	enum TrackType {
+	enum TrackType : uint8_t {
```

Comment:
I'm not sure how much this improves, the original data type would have been `int` but the alignment and other factors prevent significant optimization generally

Reply:
This was done in order to reduce the size of the `Track` from 40 to 32 bytes. You can check using sizeof.